api/v3/FinancialTrxn.php

```diff
@@ -39,7 +39,10 @@
  * @return array
  */
 function civicrm_api3_financial_trxn_create($params) {
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (empty($params['contribution_id']) && (empty($params['entity_table']) && empty($params['entity_id']))) {
```

Comment:
Note the big difference between what you did, functionally, and verify_mandatory is that if id is passed in verify_mandatory passes it through. The BAO function doesn't seem to handle passing in id well. I don't think the change proposed makes it worse though.